src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartDataGroupingsConfiguration.tsx

```diff
@@ -0,0 +1,216 @@
+import ChartBuilderSaveActions from '@admin/pages/release/datablocks/components/chart/ChartBuilderSaveActions';
+import { useChartBuilderFormsContext } from '@admin/pages/release/datablocks/components/chart/contexts/ChartBuilderFormsContext';
+import ChartDataGroupingForm from '@admin/pages/release/datablocks/components/chart/ChartDataGroupingForm';
+import generateDataSetLabel from '@admin/pages/release/datablocks/components/chart/utils/generateDataSetLabel';
+import { ChartOptions } from '@admin/pages/release/datablocks/components/chart/reducers/chartBuilderReducer';
+import ButtonText from '@common/components/ButtonText';
+import Effect from '@common/components/Effect';
+import { Form } from '@common/components/form';
+import Modal from '@common/components/Modal';
+import {
+  AxisConfiguration,
+  MapConfig,
+  MapDataSetConfig,
+  dataGroupingTypes,
+} from '@common/modules/charts/types/chart';
+import { LegendConfiguration } from '@common/modules/charts/types/legend';
+import createDataSetCategories from '@common/modules/charts/util/createDataSetCategories';
+import expandDataSet from '@common/modules/charts/util/expandDataSet';
+import generateDataSetKey from '@common/modules/charts/util/generateDataSetKey';
+import getDataSetCategoryConfigs from '@common/modules/charts/util/getDataSetCategoryConfigs';
+import { FullTableMeta } from '@common/modules/table-tool/types/fullTable';
+import { TableDataResult } from '@common/services/tableBuilderService';
+import { Formik } from 'formik';
+import isEqual from 'lodash/isEqual';
+import React, { ReactNode, useCallback, useMemo, useState } from 'react';
+
+const formId = 'chartDataGroupingsConfigurationForm';
+
+interface FormValues {
+  dataSetConfigs: MapDataSetConfig[];
+}
+
+interface Props {
+  axisMajor: AxisConfiguration;
+  buttons?: ReactNode;
+  data: TableDataResult[];
+  legend: LegendConfiguration;
+  map?: MapConfig;
+  meta: FullTableMeta;
+  options: ChartOptions;
+  onChange: (dataSetConfigs: MapDataSetConfig[]) => void;
+  onSubmit: (dataSetConfigs: MapDataSetConfig[]) => void;
+}
+
+const ChartDataGroupingsConfiguration = ({
+  axisMajor,
+  buttons,
+  data,
+  legend,
+  map,
+  meta,
+  options,
+  onChange,
+  onSubmit,
+}: Props) => {
+  const [editDataSetConfig, setEditDataSetConfig] = useState<{
+    dataSetConfig: MapDataSetConfig;
+    unit: string;
+  }>();
+  const { updateForm, submitForms } = useChartBuilderFormsContext();
+
+  const handleChange = useCallback(
+    (values: FormValues) => onChange(values.dataSetConfigs),
+    [onChange],
+  );
+
+  const handleMount = useCallback(
+    (values: FormValues) => onChange(values.dataSetConfigs),
+    [onChange],
+  );
```

Comment:
Not a dealbreaker but could just re-use `handleChange` for the Effect's `onMount`